static/app/views/integrationPipeline/pipelineView.tsx

```diff
@@ -24,27 +20,27 @@ type Props = {
   [key: string]: any;
 };
 
-export default class PipelineView extends React.Component<Props> {
-  componentDidMount() {
-    // update the title based on our mappings
-    const title = this.mapping[1];
-    document.title = title;
-  }
-  get mapping() {
-    const {pipelineName} = this.props;
-    const mapping = pipelineMapper[pipelineName];
-    if (!mapping) {
-      throw new Error(`Invalid pipeline name ${pipelineName}`);
-    }
-    return mapping;
-  }
-  render() {
-    const Component = this.mapping[0];
-    return (
-      <ThemeAndStyleProvider>
-        <Indicators className="indicators-container" />
-        <Component {...this.props} />
-      </ThemeAndStyleProvider>
-    );
+/**
+ * This component is a wrapper for specific pipeline views for integrations
+ */
+function PipelineView({pipelineName, ...props}: Props) {
+  const mapping = pipelineMapper[pipelineName];
+
+  if (!mapping) {
+    throw new Error(`Invalid pipeline name ${pipelineName}`);
   }
+
+  const [Component, title] = mapping;
+
+  // Set the page title
+  useEffect(() => void (document.title = title), [title]);
```

Comment:
Should we be using `react-document-title` here? Could be done separately.

Reply:
Yeah can updated after